### ISSUE TITLE
Conflict with autoCreateSubnetworks

### ISSUE THREAD

**Issue by cxhercules** [cloudfoundry-incubator/bosh-google-cpi-release#238]
I was following the documentation once again to deploy an environment and ran into this issue when trying to deploy cloud foundry:

```
Error applying plan:

2 error(s) occurred:

* google_compute_subnetwork.cf-compilation-subnet-1: Error creating subnetwork: googleapi: Error 400: Invalid value for field 'resource.ipCidrRange': '10.200.0.0/16'. Extended subnetworks in auto subnet mode networks cannot overlap with 10.128.0.0/9., invalid
* google_compute_address.cf-tcp: Quota 'STATIC_ADDRESSES' exceeded.  Limit: 1.0
```

The one I want to focus on is the first one that has a conflict with the auto subnet range 10.128.0.0/9. 

These are the ranges, and  yes they would conflict:
***10.128.0.0/9 - auto assign***
10.128.0.1 - 10.255.255.254

***10.200.0.0/16 - compilation_subnet***
10.200.0.1 - 10.200.255.254

To get around this I just changed the compilation range to be 10.120.0.0/16. 

## Hash is current master
https://github.com/cloudfoundry-incubator/bosh-google-cpi-release/blob/7bcae95f5a070bbc12ced2faf58f83d023e1ac95/docs/cloudfoundry/cloudfoundry.tf#L48

## Google Network reference - Last updated Oct 18, 2017
https://cloud.google.com/compute/docs/reference/latest/networks

| Property name | Value | Description |
| --- | --- | --- |
| autoCreateSubnetworks | boolean | When set to true, the network is created in "auto subnet mode". When set to false, the network is in "custom subnet mode". In "auto subnet mode", a newly created network is assigned the default CIDR of 10.128.0.0/9 and it automatically creates one subnetwork per region.  |

**Comment by johnsonj** [comment]
Thank you for checking this out and digging in @cxhercules ! Really great bug report/doc.

I've fixed/merged this as #236 which disables the auto networks. It's possible as we get more GCP regions that it would become an issue in the future. 

I'll work on getting `develop` released so this doesn't trip anyone else up.

**Comment by cxhercules** [comment]
Thanks I will close pull request I put in then thanks!

**Comment by cxhercules** [comment]
@johnsonj Although as you said this would be issue in future, also thinking the api would still complain since the range for compilation systems is still within auto subnet range. I think declaring those outside auto subnet range would fix this for now. Let me know what you think.

**Comment by johnsonj** [comment]
The referenced fix (#236) disabled the auto creation of subnets so it can't conflict. You may need to delete your infrastructure and re-create it for the change to apply. You could also manually delete the auto created subnets. 

`autoCreateSubnetworks` is nice for convenience/the default network. It's an anti pattern when you're manually defining subnets as we are in the CPI documentation. It's very easy to create subnets if you're creating a network in terraform.

**Comment by cxhercules** [comment]
I gotcha ok then thank you!
